Lib/Protocols/IdHL7.pas

```diff
@@ -1067,26 +1243,34 @@ procedure TIdHL7.StartClient;
 procedure TIdHL7.StopClient;
 var
   LFinished: Boolean;
-  LWaitStop: LongWord;
+//  LWaitStop: LongWord;
+  LStartTime : Cardinal;
 begin
   assert(Assigned(Self));
   assert(Assigned(FLock));
   FLock.Enter;
   try
-    FClientThread.Terminate;
-    FClientThread.FClient.Disconnect;
-    FClientThread.FCloseEvent.SetEvent;
+	  If Assigned(FClientThread) Then
+	  Begin
+
+		FClientThread.Terminate;
+		FClientThread.FClient.Disconnect;
+		FClientThread.FCloseEvent.SetEvent;
+      End
+    Else
+      InternalSetStatus(isStopped, 'Stopped');
   finally
     FLock.Leave;
     end;
-  LWaitStop := Ticks + 5000;
+  LStartTime := GetTickCount;
+//  LWaitStop := Ticks + 5000;
   repeat
     LFinished := (GetStatus = IsStopped);
     if not LFinished then
       begin
       IndySleep(10);
       end;
-  until LFinished or (Ticks > LWaitStop);
+  Until LFinished Or (GetTickDiff(LStartTime,GetTickCount) > WAIT_STOP);

```

Comment:
`GetTickDiff()` requires use of `IdGlobal.Ticks()`, not `GetTickCount()`

Reply:
Changed as requested
https://github.com/IndySockets/Indy/pull/442/commits/7bd8d29be4574608893391b97fcb794d959042b9